index.bs

```diff
@@ -318,6 +339,285 @@ NOTE: The size of the last layer can be determined by subtracting the sum of the
+    <tr><td>2</td><td>32</td></tr>
+    <tr><td>3</td><td>64</td></tr>
+  </tbody>
+</table>
+
+The result of any computation underflowing or overflowing the intermediate bit depth is replaced by -2<sup><code>num_bits</code>-1</sup> and 2<sup><code>num_bits</code>-1</sup>-1, respectively. Encoder implementations should not create files leading to potential computation underflow or overflow. Decoder implementations shall check for computation underflow or overflow and adapt the results accordingly. Computations with operands of negative values use the two’s-complement representation.
+
+<dfn noexport>token_count</dfn> is the expected number of tokens to read.
+
+<dfn noexport>token</dfn> determines the type of the operand (constant or input image item sample) or the operator (how to transform one or two operands into the result). See Table 2. Other values than the ones in the table below are reserved.
+
+<table class="data">
+  <caption style="caption-side:bottom">
+    Table 2 - Meaning of the value of <code>token</code>.
+  </caption>
+  <thead>
+    <tr>
+      <th>Value of <code>token</code></th>
+      <th>Name</th>
+      <th>Input count</th>
+      <th>Meaning before pushing to the stack</th>
+      <th>Value pushed to the stack<br>(<math><mi>L</mi></math> and <math><mi>R</mi></math> being operands popped from the stack for operators)</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>0</td>
+      <td>constant operand</td>
+      <td>0</td>
+      <td><math><msup><mn>2</mn><mrow><mi>bit_depth</mi><mo>+</mo><mn>3</mn></mrow></math> bits from the stream read as a signed integer.</td>
+      <td>constant value</td>
+    </tr>
+    <tr>
+      <td>1</td>
+      <td>sample operand</td>
+      <td>0</td>
+      <td>Sample value from the <code>input_image_item_index</code><sup>th</sup> input image item.</td>
+      <td>input image item sample value</td>
+    </tr>
+    <tr>
+      <td>2</td>
+      <td>negate operator</td>
+      <td>1</td>
+      <td>Negation of the left operand.</td>
+      <td><math><mo>-</mo><mi>L</mi></math></td>
+    </tr>
+    <tr>
+      <td>3</td>
+      <td>absolute value operator</td>
+      <td>1</td>
+      <td>Absolute value of the left operand.</td>
+      <td><math><mo>|</mo><mi>L</mi><mo>|</mo></math></td>
+    </tr>
+    <tr>
+      <td>4</td>
+      <td>not operator</td>
+      <td>1</td>
+      <td>Bitwise complement of the operand.</td>
+      <td><math><mo>¬</mo><mi>L</mi></math></td>
+    </tr>
+    <tr>
+      <td>5</td>
+      <td>bsr operator</td>
+      <td>1</td>
+      <td>0-based index of the most significant set bit of the left operand if the left operand is strictly positive, zero otherwise.</td>
+      <td>
+        <math>
+          <mo>{</mo>
+          <mtable>
+            <mtr><mtd><mn>0</mn></mtd><mtd><mo>if</mo><mi>L</mi><mo>≤</mo><mn>0</mn></mtd></mtr>
+            <mtr><mtd>truncate<mo>(</mo><msub><mo>log</mo><mn>2</mn></msub><mi>L</mi><mo>)</mo></mtd><mtd>otherwise</mtd></mtr>
```

Comment:
We may need to define the `truncate()` function.

Reply:
> We may need to define the `truncate()` function.

Not sure where. The table contains text such as "The result is truncated toward zero (integer division)" so I do not think defining `truncate()` is necessary, but I agree it would be nice to have. Is it worth creating a definition section though?

Reply:
Yes, it is common to have a section that defines the mathmetical symbols and functions used in the spec. If `truncate()` is the only function we need to define, it is not necessary to add a definition section.

I found that there is a `trunc()` function in the Standard C Library:
https://en.cppreference.com/w/c/numeric/math/trunc

So I think it is well known that `truncate() `means truncating towards 0, i.e., discarding the fractional part after the decimal point. Let's ignore this issue.